exchanges/ftx/ftx_test.go

```diff
@@ -0,0 +1,1007 @@
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = f.GetTriggerOrderHistory(spotPair, time.Unix(1559901511, 0), time.Unix(1559881511, 0), order.Buy.Lower(), "stop", "1")
+	if err == nil {
+		t.Error(err)
+	}
+}
+
+func TestOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test, either api keys or manipulaterealorders isnt set correctly")
+	}
+	_, err := f.Order(spotPair, order.Buy.Lower(), "limit", "", "", "", "", 0.0001, 500)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestTriggerOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test, either api keys or manipulaterealorders isnt set correctly")
+	}
+	_, err := f.TriggerOrder(spotPair, order.Buy.Lower(), order.Stop.Lower(), "", "", 500, 0.0004, 0.0001, 0)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestDeleteOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test, either api keys or manipulaterealorders isnt set correctly")
+	}
+	_, err := f.DeleteOrder("testing123")
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestDeleteOrderByClientID(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test, either api keys or manipulaterealorders isnt set correctly")
+	}
+	_, err := f.DeleteOrderByClientID("clientID123")
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestDeleteTriggerOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test, either api keys or manipulaterealorders isnt set correctly")
+	}
+	_, err := f.DeleteTriggerOrder("triggerOrder123")
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetFills(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := f.GetFills(spotPair, "", time.Time{}, time.Time{})
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = f.GetFills(spotPair, "", time.Unix(1559881511, 0), time.Unix(1559901511, 0))
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = f.GetFills(spotPair, "", time.Unix(1559901511, 0), time.Unix(1559881511, 0))
+	if err == nil {
+		t.Error(err)
+	}
+}
+
+func TestGetFundingPayments(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := f.GetFundingPayments(time.Time{}, time.Time{}, "")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = f.GetFundingPayments(time.Unix(1559881511, 0), time.Unix(1559901511, 0), "")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = f.GetFundingPayments(time.Unix(1559901511, 0), time.Unix(1559881511, 0), "")
+	if err == nil {
+		t.Error(err)
+	}
+}
+
+func TestListLeveragedTokens(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip()
+	}
+	_, err := f.ListLeveragedTokens()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetTokenInfo(t *testing.T) {
```

Comment:
`  TestGetTokenInfo: ftx_test.go:450: json: cannot unmarshal array into Go struct field TokenInfo.result of type ftx.LeveragedTokensData`